### PR TITLE
Adaptation for oneclick order cart rules (#99)

### PR DESCRIPTION
**Pull request by mehdichaouch**
https://one-click-api.readme.io/reference#r%C3%A8gles-paniers

### CODE REVIEW

File: src/Classes/OneClickItem.php [hunk 1]
```diff
@@ -163,15 +107,23 @@ public function toArray()
     {
         $oneClickItem = array(
             'reference' => $this->reference,
+            'title' => $this->title,
+            'amount' => $this->amount->toArray(),
```
Comment (feyromain) [inline src/Classes/OneClickItem.php]: Add check field is OystPrice ?

Comment (mehdichaouch) [reply src/Classes/OneClickItem.php]: The check is already made in constructor
image

Comment (feyromain) [reply src/Classes/OneClickItem.php]: Good

File: src/Classes/OneClickOrderCartEstimate.php [hunk 2]
```diff
@@ -207,8 +289,10 @@ public function toArray()
         $oneClickShipmentCalculation = array(
```
Comment (feyromain) [inline src/Classes/OneClickOrderCartEstimate.php]: Replace $oneClickShipmentCalculation -> $oneClickOrderCartEstimate

File: src/Classes/OneClickOrderCartEstimate.php [hunk 3]
```diff
@@ -207,8 +289,10 @@ public function toArray()
...
             'shipments' => $oystCollectionHelper->collectionToArray($this->shipments),
             'items' => $oystCollectionHelper->collectionToArray($this->items),
-            'message' => $this->message,
             'order_amount' => $this->orderAmount instanceof OystPrice ? $this->orderAmount->toArray() : array(),
+            'free_items' => $oystCollectionHelper->collectionToArray($this->freeItems),
+            'merchant_discounts' => $oystCollectionHelper->collectionToArray($this->merchantDiscounts),
+            'message' => $this->message,
         );
 
         return $oneClickShipmentCalculation;
```
Comment (feyromain) [inline src/Classes/OneClickOrderCartEstimate.php]: Replace $oneClickShipmentCalculation -> $oneClickOrderCartEstimate

File: src/Classes/OneClickItem.php [hunk 4]
```diff
@@ -13,145 +13,89 @@
...
+     * @param string $title
+     * @param OystPrice $amount
+     * @param int $quantity
      */
-    public function getAmountOriginal()
+    public function __construct($reference, $title, OystPrice $amount, $quantity)
     {
-        return $this->amountOriginal;
+        $this->reference = (string)$reference;
+        $this->title= (string)$title;
```
Comment (feyromain) [inline src/Classes/OneClickItem.php]: Remove title. Title is not exist for items.

File: src/Classes/OneClickOrderCartEstimate.php [hunk 5]
```diff
@@ -167,6 +189,66 @@ public function setOrderAmount(OystPrice $orderAmount)
...
+     *
+     * @return $this
+     */
+    public function setFreeItems($freeItems)
+    {
+        $this->freeItems = $freeItems;
+
+        return $this;
+    }
+
```
[the remark this answers is not in the document]
Comment (mehdichaouch) [reply src/Classes/OneClickOrderCartEstimate.php]: done.

**Pull request merged**